packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/Share.kt

```diff
@@ -0,0 +1,209 @@
+package dev.fluttercommunity.plus.share
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import androidx.core.content.FileProvider
+import java.io.File
+import java.io.IOException
+
+/**
+ * Handles share intent. The `context` and `activity` are used to start the share
+ * intent. The `activity` might be null when constructing the [Share] object and set
+ * to non-null when an activity is available using [.setActivity].
+ */
+internal class Share(private val context: Context?, private var activity: Activity?) {
+  private val providerAuthority: String by lazy {
+    getContext().packageName + ".flutter.share_provider"
+  }
+
+  private val shareCacheFolder: File
+    get() = File(getContext().cacheDir, "share_plus")
+
+  private fun getContext(): Context {
+    if (activity != null) {
+      return activity!!
+    }
+    if (context != null) {
+      return context
+    }
+    throw IllegalStateException("Both context and activity are null")
+  }
+
+  /**
+   * Sets the activity when an activity is available. When the activity becomes unavailable, use
+   * this method to set it to null.
+   */
+  fun setActivity(activity: Activity?) {
+    this.activity = activity
+  }
+
+  fun share(text: String, subject: String?) {
+    val shareIntent = Intent().apply {
+      action = Intent.ACTION_SEND
+      type = "text/plain"
+      putExtra(Intent.EXTRA_TEXT, text)
+      putExtra(Intent.EXTRA_SUBJECT, subject)
+    }
+
+    val chooserIntent = Intent.createChooser(shareIntent, null /* dialog title optional */)
```

Comment:
Tried to preserve all the original code comments.
I believe at a later point we could provide additional parameter to let users set the title for this chooser.